python/cudf/cudf/core/column/column.py

```diff
@@ -1248,9 +1254,8 @@ def corr(self, other):
         )
 
     def nans_to_nulls(self):
-        if self.dtype.kind == "f":
-            col = self.fillna(np.nan)
-            newmask = libcudf.transform.nans_to_nulls(col)
+        if self.dtype.kind == "f" and libcudf.unary.is_nan(self).any():
```

Comment:
Isn't this `is_nan.any()` check more expensive than just doing `nans_to_nulls`? `is_nan.any()` would cause unnecessary synchronization.

Reply:
Had this chat privately but posting here for visibility, there seems to be huge performance hit when we use `is_nan.any()`, so removed it: 

```python

In [1]: import cudf

In [2]: import numpy as np

In [3]: s = cudf.Series([1, 2, np.nan, None, np.nan, 324, 123, 45] * 10_00_000_00, nan_as_null=False)


# With is_nan.any() check enabled

In [5]: %timeit s.nans_to_nulls()
84.4 ms ± 5.55 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [6]: %timeit s.nans_to_nulls()
85.3 ms ± 4.44 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)




# Without is_nan.any() check


In [4]: %timeit s.nans_to_nulls()
43.6 ms ± 4.72 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [5]: %timeit s.nans_to_nulls()
44.9 ms ± 5.1 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```